### DISCUSSION TITLE
Cinematic shot composition

### DISCUSSION THREAD

**Discussion by ZenithStar** [ramokz/phantom-camera#279] [Show and tell]
I started writing some logic to compute camera distance/angle based on parameters relative to a mesh (e.g., a character's face mesh) orientation and AABB. This is integrated as a [Dialogic Event+Subsystem](https://docs.dialogic.pro/creating-extensions.html).
![image](https://github.com/ramokz/phantom-camera/assets/12200619/a3298079-f00b-4d7b-b67b-470c974b3068)
[2024-04-26 10-23-30.webm](https://github.com/ramokz/phantom-camera/assets/12200619/2a7f8d5b-b9f1-41ce-918a-259c6920aaef)

So far, it's just a basic proof of concept with a single mode implemented, but it's sufficient for my current use cases. In the near future, I plan to implement a few more as I need to use them, as well as adding DoF and frustum parameters, when those become available. Based on terminology from [this guide series](https://www.youtube.com/watch?v=AyML8xuKfoc&list=PLEzQZpmbzckV0_a2QCO2qF9Yfe-LKSDha).

If y'all think these features make sense for this repo, I can add them to PhantomCamera. They would probably exist as set of static functions that instantiate new PhantomCameras based on different sets of parameters.

**Comment by tiloc** [comment]
Awesome, great minds think alike! I watched StudioBinder also a while ago and tried to use more cinematic terms, rather than a very engineering/math driven way of thinking. I created a little scene I call a "Cinematic Rig" that I can make a child of my characters, that contains reference markers for various height levels, angels and distances. This allows me to say "make the camera look over the left shoulder of my character, point it far into the distance, and make it eye level". This is very simplistic right now, haven't integrated it yet with Phantom Camera, only using it with plain Camera3D. I was also thinking about somehow integrating it with the skeleton of my character, to dynamically adapt to their height.
